### ISSUE TITLE
Prevent click event on child element when dragging

### ISSUE THREAD

**Issue by alexcoroza** [davidcetinkaya/embla-carousel#24]
I created a carousel and added my custom clickable elements inside. The problem is when I drag the carousel and then I mouseup, click event is triggered which is not the desired action.

**Comment by davidcetinkaya** [comment]
Hi Alex (@alexcoroza),

Thanks for taking time to open this issue. Would you mind **forking** this [CodeSandbox](https://codesandbox.io/s/oyols) to demonstrate the issue and what you want to achieve? It's much easier for me to troubleshoot and help you that way. Thank you for your patience.

Kindly,
David

**Comment by alexcoroza** [comment]
Oooh, Didnt expect this, you responded so quick. Here is the example scenario.

[https://codesandbox.io/s/embla-carousel-loop-false-4e5rp]

When you drag to change slides and then you mouseup on top of a slide, the click event is triggered. Actually Im using this library in my angular app so the sample codes in sandbox is quite different(but they show the same problem).

I tried many javascript drag scroll carousel libraries they have all this kind of issue except for the ngx-drag-scroll(though its laggy). As of  now I'm just showing an inivisible overlay whenever .on('scrolling') is called then hide it when finished, just to prevent the child element's click event

**Comment by davidcetinkaya** [comment]
Hello Alex (@alexcoroza),

Thank you for the sample code 👍. Just want to make sure that I understand you correctly before we start discussing how to solve this issue:

1. Do you want to prevent the click event from triggering **if dragging has occurred**?
2. Or do you actually want to prevent the click event from triggering **if carousel is scrolling**?

All the best,
David

**Comment by alexcoroza** [comment]
Yes the question 1. I just want to prevent the click event only when dragging.

**Comment by davidcetinkaya** [comment]
Hello again Alex (@alexcoroza),

Thank you for getting back to me. Maybe we should consider implementing a `staticClick event` or similar? I'm thinking that this event **would only fire** when:

- User clicks on the carousel
- No dragging occurs

What's your thoughts? Or do you have any other ideas?

Best,
David

**Comment by davidcetinkaya** [comment]
Hello again Alex (@alexcoroza),

Which version of Embla are you using? Is it the [Vanilla JS version](https://github.com/davidcetinkaya/embla-carousel) or the [React version](https://github.com/davidcetinkaya/embla-carousel-react).

Kindly,
David

**Comment by davidcetinkaya** [comment]
Hello Alex (@alexcoroza),
I'm happy to announce that I've released a click check for Embla: 

```javascript
embla.clickAllowed()
```

Please read about it in the [release description](https://github.com/davidcetinkaya/embla-carousel/releases/tag/v2.6.0). You can see it in action in this [CodeSandbox](https://codesandbox.io/s/embla-carousel-clickallowed-woxr1). This method is an **opt-in feature**. Here is how it works: It returns `false` if:

**Mouse Pointers**

- Mouse drags the carousel.

**Touch Pointers**
- Touch drags the carousel.
- Carousel is in motion (i.e. is scrolling).

Otherwise it will return `true`. I'd very much appreciate your feedback on this and if it solves your issue. Thank you in advance!

Kindly,
David

**Comment by davidcetinkaya** [comment]
Hi Alex (@alexcoroza),

I’m curious if you’ve had the chance to try this yet?

Kindly,
David

**Comment by davidcetinkaya** [comment]
Hello again Alex (@alexcoroza),
I'm closing this due to lack of response. Feel free to reopen this issue if you don't think it's resolved.

Thank you for your feature request!
David

**Comment by coopemor** [comment]
Hey @davidcetinkaya this is exactly what I need but I'm struggling to figure out how to implement it while using the react version. Any chance you could provide a bit of guidance?

Thanks so much for you hard work in this, it has the smoothest animations I've found for a carousel library!

**Comment by davidcetinkaya** [comment]
Hi @coopemor,

I would be happy to help. In order to do so I need additional details. Are you using React hooks or class based components?

Kindly,
David

**Comment by coopemor** [comment]
Amazing! Thank you @davidcetinkaya I'm using hooks in my design.

**Comment by davidcetinkaya** [comment]
Thanks for the additional information @coopemor. For details about how `embla.clickAllowed()` works, please read the [documentation here](https://github.com/davidcetinkaya/embla-carousel#clickallowed).

So here's **one way to do it in React**:

```jsx
import React, { useCallback } from "react";
import { useEmblaCarousel } from "embla-carousel-react";

const EmblaCarouselComponent = ({ children }) => {
  const [EmblaCarouselReact, embla] = useEmblaCarousel({ loop: false });

  const onSlideClick = useCallback(
    slideIndex => () => {
      if (embla && embla.clickAllowed()) {
        // Do your magic
        console.log(`You clicked on slide with index ${slideIndex}`);
      }
    },
    [embla]
  );

  return (
    <div className="embla">
      <EmblaCarouselReact className="embla__viewport">
        <div className="embla__container">
          {children.map((Child, index) => (
            <div
              key={index}
              onClick={onSlideClick(index)}
              className="embla__slide"
            >
              {Child}
            </div>
          ))}
        </div>
      </EmblaCarouselReact>
    </div>
  );
};

export default EmblaCarouselComponent;
```

Let me know if this helps you on the right track 🙂!

Best,
David

**Comment by davidcetinkaya** [comment]
Hello @coopemor,

Your question is related to the [Embla React package](https://github.com/davidcetinkaya/embla-carousel-react/), so I decided to create a new issue there. Just [follow this link](https://github.com/davidcetinkaya/embla-carousel-react/issues/26) and we'll continue this conversation there.

Thanks!
David
